currency_rate_update/wizards/res_currency_rate_update_wizard.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api
+
+
+class ResCurrencyRateUpdateWizard(models.TransientModel):
+    _name = 'res.currency.rate.update.wizard'
+    _description = 'Currency Rate Update Wizard'
+
+    date_from = fields.Date(
+        string='Start Date',
+        required=True,
+        default=fields.Date.context_today,
+    )
+    date_to = fields.Date(
+        string='End Date',
+        required=True,
+        default=fields.Date.context_today,
+    )
+    provider_ids = fields.Many2many(
+        string='Providers',
+        comodel_name='res.currency.rate.provider',
+        column1='wizard_id',
+        column2='provider_id',
+    )
+
+    @api.multi
+    def action_update(self):
+        self.ensure_one()
+
+        self.provider_ids._update(self.date_from, self.date_to)
+
+        return {'type': 'ir.actions.act_window_close'}
```

Comment:
This is redundant

Reply:
Why? It will autoclose?

Reply:
I don't think I've added this because of nothing - when the button is pressed, I need to close the dialog window.

Reply:
But that's the default action. Have you tried without this?